### PR TITLE
SM/CUDA: Implement iface_get_device_address.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1052/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1664/ for details.

File: src/uct/sm/cma/cma_iface.c [hunk 1]
```diff
@@ -46,6 +47,7 @@ static ucs_status_t uct_cma_iface_query(uct_iface_h tl_iface,
     iface_attr->cap.put.max_zcopy      = SIZE_MAX;
     iface_attr->cap.get.max_zcopy      = SIZE_MAX;
     iface_attr->iface_addr_len         = sizeof(uct_sockaddr_process_t);
+    iface_attr->iface_addr_len         = UCT_SM_IFACE_DEVICE_ADDR_LEN;
```
Comment (shamisp) [inline src/uct/sm/cma/cma_iface.c]: where do you set uct_sm_iface_get_device_address for knem ? for some reason it is still set to ucs_empty_function_return_success

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1668/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1056/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp  pls review

File: src/uct/cuda/cuda_iface.c [hunk 2]
```diff
@@ -73,7 +75,7 @@ static ucs_status_t uct_cuda_iface_query(uct_iface_h iface,
 static uct_iface_ops_t uct_cuda_iface_ops = {
     .iface_close         = UCS_CLASS_DELETE_FUNC_NAME(uct_cuda_iface_t),
     .iface_get_address   = uct_cuda_iface_get_address,
-    .iface_get_device_address = (void*)ucs_empty_function_return_success,
+    .iface_get_device_address = uct_sm_iface_get_device_address,
```
Comment (shamisp) [inline src/uct/cuda/cuda_iface.c]: you should not reference uct_sm_iface_get_device_address sm function from cuda tl

File: src/uct/cuda/cuda_iface.c [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #include "cuda_pd.h"
 #include "cuda_ep.h"
 
+#include <uct/sm/base/sm_iface.h>
```
Comment (shamisp) [inline src/uct/cuda/cuda_iface.c]: yeah...this is kinda ugly

Comment (shamisp) [inline src/uct/cuda/cuda_iface.c]: the common code should go up to tl/base ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1059/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1672/ for details.

**Comment by yosefe** [comment]
@shamisp fixed - removed device address from cuda at all, for now.

**Comment by shamisp** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1673/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1674/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1675/ for details.

**Pull request merged**